### PR TITLE
drop several unneeded PL_API_VER checks

### PR DESCRIPTION
**Pull request by Dudemanguy**
Also a bugfix for the pl_cache not being destroyed on uninit.

### CODE REVIEW

**Comment by Dudemanguy** [comment]
Changed the third commit to rewrite the cache usage to use the newer API which fixes a few bugs in the process.

**Comment by kasper93** [comment]
You could also remove `PL_HAVE_LAV_HDR`. On the libavutil version that we require it is always available.

``` C
#if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(56, 25, 100)
# define PL_HAVE_LAV_HDR
#endif
```

``` meson
libavutil = dependency('libavutil', version: '>= 56.70.100')
```

I nuked this part of code in my other PR completely, but yours probably will be merged sooner, so you can include this if you like.

**Comment by Dudemanguy** [comment]
>PL_HAVE_LAV_HDR

Didn't even notice that. Can `PL_HAVE_LAV_FILM_GRAIN` also be dropped?

File: video/out/vo_gpu_next.c [hunk 1]
```diff
@@ -1612,6 +1528,61 @@ static char *get_cache_file(struct priv *p)
...
+        if (!same_cache) {
+            FILE *cache = fopen(icc_cache, "wb");
+            if (cache) {
+                pl_cache_save_file(p->icc_cache, cache);
+                fclose(cache);
+            }
+        }
+        talloc_free(icc_cache);
+    }
+}
```
Comment (kasper93) [inline video/out/vo_gpu_next.c:1583]: Maybe add some error logging here? "failed to load cache" "failed to save cache" and so on. Maybe even trace ones when  it loads successfully with size.

File: video/out/vo_gpu_next.c [hunk 2]
```diff
@@ -1696,31 +1658,25 @@ static int preinit(struct vo *vo)
...
     pthread_mutex_init(&p->dr_lock, NULL);
 
-#if PL_API_VER >= 320
-    p->cache = pl_cache_create(pl_cache_params(
+    p->shader_cache = pl_cache_create(pl_cache_params(
         .log = p->pllog,
-        .max_object_size =  1 << 20, //  1 MB
-        .max_total_size  = 10 << 20, // 10 MB
+        .max_object_size =  1 << 20, //  1 MiB
+        .max_total_size  = 10 << 20, // 10 MiB
```
Comment (kasper93) [inline video/out/vo_gpu_next.c:1664]: This should be 50 MiB and without per object size limit. https://code.videolan.org/videolan/libplacebo/-/commit/ea91056682d1117a2b70b8a2c17773180aa3ac2c

gamut mapping luts are bigger, than 1 MiB and overall 10 MiB is not enough. 0/50 MiB is reasonable limit.

File: video/out/vo_gpu_next.c (hunk 1 above)
Comment (Dudemanguy) [reply video/out/vo_gpu_next.c:1583]: Well libplacebo already logs successful saved/loaded objects, but yes we can log a warning if it fails.

Comment (kasper93) [reply video/out/vo_gpu_next.c:1583]: At your discretion. I noticed a lot of I/O without logging, that generally don't end well.

File: video/out/vo_gpu_next.c (hunk 2 above)
Comment (Dudemanguy) [reply video/out/vo_gpu_next.c]: Sure.

File: video/out/vo_gpu_next.c (hunk 1 above)
Comment (Dudemanguy) [reply video/out/vo_gpu_next.c:1591]: My log messages are not very creative but I tried something.

**Review by kasper93 — APPROVED** [review]

**Pull request merged**